src/app/core/services/scheduling.service.ts

```diff
@@ -220,6 +242,21 @@ export class SchedulingService {
       schedule = schedule.concat(tmpSchedule)
       scheduleLength = schedule.length
     }
+    // NOTE: Check for completed tasks
+    this.completedTasks.map(d => {
+      const index = schedule.findIndex(
```

Comment:
I think we can branch here according to 2 cases. One where offset is changed and one where it is same. The offset will usually be same on protocol change and hence just comparing the name and timestamp should be sufficient in that case thus avoiding the calculations. The other case will include all the above code and currentMidnight and prevMidnight initializations. What do you think?

Reply:
Ok. I've made the changes.